src/sqlfluff/dialects/dialect_sqlite.py

```diff
@@ -93,3 +94,83 @@ class InsertStatementSegment(BaseSegment):
             Sequence("DEFAULT", "VALUES"),
         ),
     )
+
+
+class ColumnConstraintSegment(BaseSegment):
```

Comment:
This looks fairly similar to the `ColumnConstraintSegment` in ANSI dialect. Did you consider one or both of:
* Modifying the ANSI dialect, if the changes are minor (especially if other dialects in the real world use the same syntax)
* Using the `copy/insert/before` technique seen in Postgres and other dialects to extend rather than duplicate existing syntax

Reply:
Thanks for the feedback! The code was redundant for just deferrable section addition, so I applied the latter solution as you mentioned. [fa764e4](https://github.com/sqlfluff/sqlfluff/pull/3757/commits/fa764e44882954a673a9f552ac7748c74145d469)